### PR TITLE
Fixed various DNA code issues that allowed base monkeys to be humanized

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Added a NO_DNA_SCRAMBLE check to the actual inactive mutation shuffling action.

Added a mutadone check to the remove individual mutation proc.

Added arguments to inflate_gib().

Syndicate Monkeys will now slowly explode via the above proc if they're species-changed.

This PR will not compile due to a bug on master.
## Why It's Good For The Game

> Added a NO_DNA_SCRAMBLE check to the actual inactive mutation shuffling action.

I added this to the wrong proc. Turns out scramble_dna() is unrelated to Scramble DNA. Yeah.

> Added a mutadone check to the remove individual mutation proc.

Someone forgor this and it became an issue when TRAIT_BORN_MONKEY was added, which calls this proc instead of remove_mutation_groups on ingestion, which does have the parameter.

> Added parameters to inflate_gib().

Default is the same as it used to be.

> Syndicate Monkeys will now slowly explode via the above proc if they're species-changed.

I love seeing people use my syndicate monkeys but more often than not they or the monkey immediately fuck off to become a human in genetics, and holy shit that riles me up hard. This item is not meant to give you a free human buddy. It's meant to give you a free monkey buddy. Originally I didn't worry too much about the crew finding out ways to humanize, but I changed my mind after seeing how one-note players are. I don't want another lame cheese exploit causing people to use this as cheap reinforcements.

The uplink item's desc has been updated to fit this new information.

You can still brain transfer, if you want to lose the clever mutation and have a human that can't use machines.
## Changelog
:cl:
fix: We found out that the money meant to purchase mutation vaccines for Syndicate Monkey agents against mutation was being embezzled to buy Space Yachts. We've repurposed their biomass towards a new and improved vaccine, which has the minor side effect of causing allergies if their species type is forcibly altered.
/:cl:

### CODE REVIEW

File: code/datums/dna.dm [hunk 1]
```diff
@@ -150,7 +150,9 @@ GLOBAL_LIST_INIT(total_uf_len_by_block, populate_total_uf_len_by_block())
 	SEND_SIGNAL(holder, COMSIG_CARBON_GAIN_MUTATION, mutation_type, class)
 	return force_give(new mutation_type (class, time, copymut = mutation))
 
-/datum/dna/proc/remove_mutation(mutation_type)
+/datum/dna/proc/remove_mutation(datum/mutation/human/mutation_type, mutadone)
+	if((mutation_type.mutadone_proof && mutadone))
```
Comment (ninjanomnom) [inline code/datums/dna.dm:154]: Shouldn't you be doing this check after get_mutation()?

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/datums/dna.dm (hunk 1 above)
Comment (ninjanomnom) [reply code/datums/dna.dm:154]: mutation_type is a type, not an instanced object. But even if you used initial here to grab the type's value, you're checking the value of the mutation you pass in and not what's actually in the dna.

Comment (carlarctg) [reply code/datums/dna.dm:154]: guh

File: code/datums/dna.dm [hunk 2]
```diff
@@ -150,9 +150,16 @@ GLOBAL_LIST_INIT(total_uf_len_by_block, populate_total_uf_len_by_block())
...
+
+	var/datum/mutation/human/actual_mutation = get_mutation(mutation_type)
+
+	// Check that it exists first before trying to remove it with mutadone
+	if((actual_mutation && (actual_mutation.mutadone_proof && mutadone)))
+		return FALSE
+
 	SEND_SIGNAL(holder, COMSIG_CARBON_LOSE_MUTATION, mutation_type)
-	return force_lose(get_mutation(mutation_type))
+	return force_lose(actual_mutation || mutation_type)
```
Comment (ninjanomnom) [inline code/datums/dna.dm:162]: This proc doesnt accept a type as an argument

File: code/datums/dna.dm [hunk 3]
```diff
@@ -150,9 +150,16 @@ GLOBAL_LIST_INIT(total_uf_len_by_block, populate_total_uf_len_by_block())
...
 	return force_give(new mutation_type (class, time, copymut = mutation))
 
-/datum/dna/proc/remove_mutation(mutation_type)
+/datum/dna/proc/remove_mutation(datum/mutation/human/mutation_type, mutadone)
+
+	var/datum/mutation/human/actual_mutation = get_mutation(mutation_type)
+
+	// Check that it exists first before trying to remove it with mutadone
+	if((actual_mutation && (actual_mutation.mutadone_proof && mutadone)))
+		return FALSE
```
Comment (ninjanomnom) [inline code/datums/dna.dm:159]: You should early return out of the proc if get_mutation finds nothing

**Review by ninjanomnom — DISMISSED** [review]

**Pull request merged**